class-gf-razorpay.php

```diff
@@ -273,158 +451,145 @@ public function get_callback_attributes()
-    <input type='hidden' name='razorpay_payment_id' id='razorpay_payment_id'>
-    <input type='hidden' name='razorpay_signature'  id='razorpay_signature' >
-</form>
-<p id='msg-razorpay-success'  style='display:none'>
-    Please wait while we are processing your payment.
-</p>
-<p>
-    <button id='btn-razorpay'>Pay With Razorpay</button>
-    <button id='btn-razorpay-cancel' onclick='document.razorpayform.submit()'>Cancel</button>
-</p>
-EOT;
-        return $html;
     }
 
     public function is_callback_valid()
     {
         // Will check if the return url is valid
-        if (rgget('page') !== 'gf_razorpay_callback')
-        {
+        if (rgget('page') !== $this->_slug) {
             return false;
         }
 
         return true;
     }
 
-    public function generate_razorpay_order($entry, $form)
+    public function billing_info_fields()
     {
-        //gravity form method to get value of payment_amount key from entry
-        $paymentAmount = rgar($entry, 'payment_amount' );
+        $fields = array(
+            array('name' => self::CUSTOMER_FIELDS_NAME, 'label' => esc_html__('Name', 'gravityforms'), 'required' => false),
+            array('name' => self::CUSTOMER_FIELDS_EMAIL, 'label' => esc_html__('Email', 'gravityforms'), 'required' => false),
+            array('name' => self::CUSTOMER_FIELDS_CONTACT, 'label' => esc_html__('Phone', 'gravityforms'), 'required' => false),
+        );
 
-        //It will be null first time in the entry
-        if (empty($paymentAmount) === true)
-        {
-            $paymentAmount = GFCommon::get_order_total($form, $entry);
-            gform_update_meta($entry['id'], 'payment_amount', $paymentAmount);
-            $entry['payment_amount'] = $paymentAmount;
-        }
+        return $fields;
+    }
 
-        $key = $this->get_plugin_setting(self::GF_RAZORPAY_KEY);
+    public function init()
+    {
 
-        $secret = $this->get_plugin_setting(self::GF_RAZORPAY_SECRET);
+        add_filter( 'gform_notification_events', array( $this, 'notification_events' ), 10, 2 );
 
-        $api = new Api($key, $secret);
+        add_filter( 'gform_post_payment_action', array( $this, 'post_payment_action' ), 10, 2 );
 
-        $data = array(
-            'receipt'         => $entry['id'],
-            'amount'          => (int) round($paymentAmount * 100),
-            'currency'        => $entry['currency'],
-            'payment_capture' => 1
-        );
+        // Supports frontend feeds.
+        $this->_supports_frontend_feeds = true;
 
-        $razorpayOrder = $api->order->create($data);
+        parent::init();
 
-        gform_update_meta($entry['id'], self::RAZORPAY_ORDER_ID, $razorpayOrder['id']);
+    }
 
-        $entry[self::RAZORPAY_ORDER_ID] = $razorpayOrder['id'];
+    // Added custom event to provide option to chose event to send notifications.
+    public function notification_events($notification_events, $form)
```

Comment:
Can we please, just try to stop notification on form submission only, and it will only go after successful payment or failure of payment.